Cubical/Codata/Conat/Properties.agda

```diff
@@ -0,0 +1,127 @@
+{-# OPTIONS --cubical --safe --guardedness #-}
+module Cubical.Codata.Conat.Properties where
+
+open import Cubical.Data.Unit
+open import Cubical.Data.Sum
+open import Cubical.Data.Empty
+
+open import Cubical.Core.Everything
+open import Cubical.Foundations.Equiv
+
+open import Cubical.Relation.Nullary
+open import Cubical.Relation.Nullary.DecidableEq
+open import Cubical.Codata.Conat.Base
+
+Unwrap-prev : Conat′ -> Set
+Unwrap-prev  zero   = Unit
+Unwrap-prev (suc _) = Conat
+
+unwrap-prev : (n : Conat′) -> Unwrap-prev n
+unwrap-prev  zero   = _
+unwrap-prev (suc x) = x
+
+private -- tests
+  𝟘 = conat zero
+  one  = succ 𝟘
+  two  = succ one
+
+  succOne≡two : succ one ≡ two
+  succOne≡two i = two
+
+  predTwo≡one : unwrap-prev (force two) ≡ one
+  predTwo≡one i = one
+
+∞ : Conat
+force ∞ = suc ∞
+
+∞+1≡∞ : succ ∞ ≡ ∞
+force (∞+1≡∞ _) = suc ∞
+
+∞+2≡∞ : succ (succ ∞) ≡ ∞
+∞+2≡∞ = compPath (cong succ ∞+1≡∞) ∞+1≡∞
+
+-- TODO: plus for conat, ∞ + ∞ ≡ ∞
+
+conat-absurd : ∀ {y : Conat} {ℓ} {Whatever : Set ℓ} → zero ≡ suc y → Whatever
+conat-absurd eq = ⊥-elim (transport (cong diag eq) tt)
+  where
+  diag : Conat′ → Set
+  diag zero = Unit
+  diag (suc _) = ⊥
+
+module EqStable where
+  ≡-stable  : {x y : Conat} → Stable (x ≡ y)
+  ≡′-stable : {x y : Conat′} → Stable (x ≡ y)
+
+  force (≡-stable ¬¬p i) = ≡′-stable (λ ¬p → ¬¬p (λ p → ¬p (cong force p))) i
+  ≡′-stable {zero}  {zero}  ¬¬p′ = refl
+  ≡′-stable {suc x} {suc y} ¬¬p′ =
+     cong′ suc (≡-stable λ ¬p → ¬¬p′ λ p → ¬p (cong pred′′ p))
+  ≡′-stable {zero}  {suc y} ¬¬p′ = ⊥-elim (¬¬p′ conat-absurd)
+  ≡′-stable {suc x} {zero}  ¬¬p′ = ⊥-elim (¬¬p′ λ p → conat-absurd (sym p))
+
+  ≡-unique : {m n : Conat} (p q : m ≡ n) → p ≡ q
+  ≡-unique = Stable≡→isSet (λ _ _ → ≡-stable) _ _
```

Comment:
Should this be named some variation on `isSetConat`? Also isn't that type signature exactly `isSet Conat`

Reply:
Fixed